frontend/src/app/modules/challenges/challenge-creation/challenge-creation.utils.ts

```diff
@@ -0,0 +1,98 @@
+import { CategoryType } from '@shared/enums/category-type';
+import { ChallengeStep } from '@shared/enums/challenge-step';
+import { NewChallenge } from '@shared/models/challenge/new-challenge';
+import { NewChallengeVersion } from '@shared/models/challenge-version/new-challenge-version';
+import { DropdownItem } from '@shared/models/dropdown-item';
+
+export const stepData = [
+    { step: ChallengeStep.Question, isValid: false, checkValidation: false },
+    { step: ChallengeStep.Solutions, isValid: false, checkValidation: false },
+    { step: ChallengeStep.Testcases, isValid: false, checkValidation: false },
+];
+
+export function getStepIndex(step: ChallengeStep) {
+    return stepData.findIndex(s => s.step === step);
+}
+
+export function getStep(step: ChallengeStep) {
+    return stepData.find(s => s.step === step);
+}
+
+export function getStepChecking(stepType: ChallengeStep) {
+    const step = getStep(stepType);
+
+    if (step) {
+        return step.checkValidation;
+    }
+
+    return false;
+}
+
+export function stepAllowed(step: ChallengeStep) {
+    const stepIndex = getStepIndex(step);
+
+    return stepData.every((value, index) => {
+        if (index < stepIndex) {
```

Comment:
Do not work with indexes, work with order or something like this
indexes are not readable and hard to understand

This is related to all code not only this part

Reply:
Ok, no indexes anymore